internal/helpers/js_comment_utils.go

```diff
@@ -0,0 +1,44 @@
+package helpers
+
+import (
+	"errors"
+	"strings"
+)
+
+// RemoveComments removes both block and inline comments from a string
+func RemoveComments(input string) (string, error) {
+	var (
+		sb        = strings.Builder{}
+		inComment = false
+	)
+	for cur := 0; cur < len(input); cur++ {
+		peekIs := func(assert byte) bool { return cur+1 < len(input) && input[cur+1] == assert }
```

Comment:
can you move this function to above the loop? This also means needing to move the `cur := 0` above as well. I don't know how the Go compiler works but naively it might be creating a new closure for each iteration of the loop which is quite expensive when looping over a string. Lifting it up should help if that's the case.

Reply:
Yeah I'll move it out of the function entirely, it's only used three times in the functions, won't hurt passing the arguments manually three times 😄